File: shuffle/cxx/test_fisher_yates.cxx
#include "fisher_yates.h"

#include <vector>

void test_fisher_yates()
{
	std::vector<int> array(100);
	for(int i = 0; i < 100; i++)
		array[i] = i;
	fisher_yates(array);
}

void test_fisher_yates_front()
{
	std::vector<int> array(100);
	for(int i = 0; i < 100; i++)
		array[i] = i;
	fisher_yates_front(array);
}

int main()
{
	test_fisher_yates();
	test_fisher_yates_front();
	return 0;
}

File: shuffle/py3/TestFisherYates.py
import unittest
from fisher_yates import fisher_yates
from fisher_yates import fisher_yates_front

class TestFisherYates(unittest.TestCase):
    def testFisherYates(self):
        array = list(range(100))
        fisher_yates(array)

    def testFisherYatesFront(self):
        array = list(range(100))
        fisher_yates_front(array)

if __name__ == "__main__":
    unittest.main()
